elementary/monitor/api/tests/tests.py

```diff
@@ -380,10 +393,15 @@ def _get_test_metadata_from_test_result_db_row(
     def _get_test_result_from_test_result_db_row(
         test_result_db_row: TestResultDBRowSchema,
         disable_samples: bool = False,
-    ) -> Union[DbtTestResultSchema, ElementaryTestResultSchema]:
-        test_results = None
+    ) -> Optional[Union[DbtTestResultSchema, ElementaryTestResultSchema]]:
+        test_results: Optional[Union[DbtTestResultSchema, ElementaryTestResultSchema]]
+
         sample_data = test_result_db_row.sample_data if not disable_samples else None
         if test_result_db_row.test_type == "dbt_test":
+            if sample_data is not None and not isinstance(sample_data, list):
+                # Sanity check, shouldn't happen
```

Comment:
Don't really like this.
If it shouldn't happen there shouldn't be a check for it.
I think it means we're doing something wrong.

Reply:
Changing to cast